From 8d12d8104f9599dc10435f8c5ce88d1bde0ca1bc Mon Sep 17 00:00:00 2001
From: Daniel J Walsh <dwalsh@redhat.com>
Date: Mon, 28 Sep 2020 13:40:15 -0400
Subject: [PATCH] Fix handling of Ambient/Inheritable caps for non root user
Bug-Debian: https://bugs.debian.org/977717

On F33 we are getting permission denied when changing default
capabilities of containers.  I believe Ineritable and Ambient
capabilities have to match Permissited and potentially Effective.

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>
---
 pkg/specgen/generate/security.go | 3 +++
 1 file changed, 3 insertions(+)

--- a/pkg/specgen/generate/security.go
+++ b/pkg/specgen/generate/security.go
@@ -136,6 +136,7 @@
 		configSpec.Process.Capabilities.Effective = caplist
 		configSpec.Process.Capabilities.Permitted = caplist
 		configSpec.Process.Capabilities.Inheritable = caplist
+		configSpec.Process.Capabilities.Ambient = caplist
 	} else {
 		userCaps, err := capabilities.NormalizeCapabilities(s.CapAdd)
 		if err != nil {
@@ -143,6 +144,8 @@
 		}
 		configSpec.Process.Capabilities.Effective = userCaps
 		configSpec.Process.Capabilities.Permitted = userCaps
+		configSpec.Process.Capabilities.Inheritable = userCaps
+		configSpec.Process.Capabilities.Ambient = userCaps
 	}
 
 	g.SetProcessNoNewPrivileges(s.NoNewPrivileges)
